streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsKafkaClient.java

```diff
@@ -189,16 +215,43 @@ private ClientResponse sendRequest(final ClientRequest clientRequest) {
      */
     public Collection<MetadataResponse.TopicMetadata> fetchTopicsMetadata() {
 
-        final ClientRequest clientRequest = kafkaClient.newClientRequest(getBrokerId(), new MetadataRequest.Builder(null), Time.SYSTEM.milliseconds(), true, null);
+        final ClientRequest clientRequest = kafkaClient.newClientRequest(
+            getBrokerId(),
+            new MetadataRequest.Builder(null),
+            Time.SYSTEM.milliseconds(),
+            true);
         final ClientResponse clientResponse = sendRequest(clientRequest);
         if (!clientResponse.hasResponse()) {
             throw new StreamsException("Empty response for client request.");
         }
         if (!(clientResponse.responseBody() instanceof MetadataResponse)) {
-            throw new StreamsException("Inconsistent response type for internal topic metadata request. Expected MetadataResponse but received " + clientResponse.responseBody().getClass().getName());
+            throw new StreamsException("Inconsistent response type for internal topic metadata request. " +
+                "Expected MetadataResponse but received " + clientResponse.responseBody().getClass().getName());
         }
         final MetadataResponse metadataResponse = (MetadataResponse) clientResponse.responseBody();
         return metadataResponse.topicMetadata();
     }
 
+    public void checkBrokerCompatibility() {
```

Comment:
We prefer a "fail fast" approach with (small stack trace) and clean error message to the user (this also safes starting up the KafakStreams instances including consumer/producer etc.)